erpnext/public/js/controllers/taxes_and_totals.js

```diff
@@ -390,10 +390,21 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
 		this.frm.doc.grand_total = flt(tax_count
 			? this.frm.doc["taxes"][tax_count - 1].total + flt(this.frm.doc.rounding_adjustment)
 			: this.frm.doc.net_total);
-
+		var payment_status = true;
 		if(in_list(["Quotation", "Sales Order", "Delivery Note", "Sales Invoice"], this.frm.doc.doctype)) {
 			this.frm.doc.base_grand_total = (this.frm.doc.total_taxes_and_charges) ?
 				flt(this.frm.doc.grand_total * this.frm.doc.conversion_rate) : this.frm.doc.base_net_total;
+			if (this.frm.doc.is_pos && this.frm.doc.is_return) {
+				$.each(this.frm.doc['payments'] || [], function (index, data) {
+					if (payment_status) {
+						data.amount = me.frm.doc.grand_total;
```

Comment:
You have set the grand_total amount directly due to which if the original invoice has multiple mode of payments then while making sales return entry the amount of first mode of payment has been set as grand total which is wrong. For more details please check below GIF

![pos_return_issue](https://user-images.githubusercontent.com/8780500/56567181-f7b64a80-65d1-11e9-9ec7-714b41f0eb7a.gif)